### PR TITLE
MINOR: update Kafka Streams docs with 3.4 KIP information

### PR DESCRIPTION
**Pull request by mjsax**

### CODE REVIEW

File: docs/streams/upgrade-guide.html [hunk 1]
```diff
@@ -303,6 +303,35 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
       adds a new config <code>default.client.supplier</code> that allows to use a custom <code>KafkaClientSupplier</code> without any code changes.
     </p>
 
+    <h3><a id="streams_api_changes_340" href="#streams_api_changes_340">Streams API changes in 3.4.0</a></h3>
+    <p>
+      <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=186878390">KIP-770</a> deprecates
+      config <code>cache.max.bytes.buffering</code> in favor of the newly introduced config <code>statestore.cache.max.bytes</code>.
+      To improve monitoring, two new metrics <code>input-buffer-bytes-total</code> and <code>cache-size-bytes-total</code>
+      were added at the DEBUG level. Note, that the KIP is only partially implement in 3.4.0 release and config
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
      were added at the DEBUG level. Note, that the KIP is only partially implemented in the 3.4.0 release, and config
```

File: docs/streams/upgrade-guide.html [hunk 2]
```diff
@@ -303,6 +303,35 @@ <h3><a id="streams_api_changes_350" href="#streams_api_changes_350">Streams API
...
+      <code>input.buffer.max.bytes</code> is not available yet.
+    </p>
+
+    <p>
+      <a href="https://cwiki.apache.org/confluence/pages/viewpage.action?pageId=211883356">KIP-873</a> enables you to multicast
+      result records to multiple partition of downstream sink topics and adds functionality for choosing to drop result records without sending.
+      The <code>Integer StreamPartitioner.partition()</code> method is deprecated and replaced by the newly added
+      <code>Optiona&lg;Set&lt;Integer&gt;&gt;StreamPartitioner.partitions()</code> method, that allows to return a set of partitions to send the record to.
```
Comment (JimGalasyn) [inline docs/streams/upgrade-guide.html]:
```suggestion
      <code>Optiona&lg;Set&lt;Integer&gt;&gt;StreamPartitioner.partitions()</code> method, which enables returning a set of partitions to send the record to.
```

**Review by JimGalasyn — APPROVED** [review]
LGTM, with a couple of suggestions.

**Review by bbejeck — APPROVED** [review]
LGTM

**Pull request merged**